SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/GenerateSchemaAndSnapshotsCommandlet.cpp

```diff
@@ -22,75 +23,146 @@ int32 UGenerateSchemaAndSnapshotsCommandlet::Main(const FString& Args)
-	GenerateSchema(SpatialGDKEditor);
-	GenerateSnapshots(SpatialGDKEditor);
+	if (Params.Contains(MapPathsParamName))
+	{
+		FString MapNameParam = *Params.Find(MapPathsParamName);
+
+		//Spaces are disallowed in all paths, so we just check for them now and exit early if an invalid path was provided
+		if (MapNameParam.Contains(TEXT(" ")))
+		{
+			UE_LOG(LogSpatialGDKEditorCommandlet, Error, TEXT("%s argument may not contain spaces."), *MapPathsParamName);
+			return 1;
+		}
+
+		FString ThisMapName;
+		FString RemainingMapPaths = MapNameParam;
+		while (RemainingMapPaths.Split(TEXT(";"), &ThisMapName, &RemainingMapPaths))
+		{
+			GenerateSchemaAndSnapshotForPath(SpatialGDKEditor, ThisMapName);
+		}
+		//When we get to this point, one of two things is true:
+		//1) RemainingMapPaths was NEVER split, and should be interpreted as a single map name
+		//2) RemainingMapPaths was split n times, and the last map that needs to be run after the loop is still in it
+		GenerateSchemaAndSnapshotForPath(SpatialGDKEditor, RemainingMapPaths);
+	}
+	else
+	{
+		//Default to everything in the project
+		GenerateSchemaAndSnapshotForPath(SpatialGDKEditor, TEXT(""));
+	}
 
 	UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Schema & Snapshot Generation Commandlet Complete"));
 
 	return 0;
 }
 
-void UGenerateSchemaAndSnapshotsCommandlet::GenerateSchema(FSpatialGDKEditor& SpatialGDKEditor)
+void UGenerateSchemaAndSnapshotsCommandlet::GenerateSchemaAndSnapshotForPath(FSpatialGDKEditor& InSpatialGDKEditor, const FString& InPath)
 {
-	SpatialGDKEditor.GenerateSchema(
-		FSimpleDelegate::CreateLambda([]() { UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Schema Generation Completed!")); }),
-		FSimpleDelegate::CreateLambda([]() { UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Schema Generation Failed")); }),
-		FSpatialGDKEditorErrorHandler::CreateLambda([](FString ErrorText) { UE_LOG(LogSpatialGDKEditorCommandlet, Error, TEXT("%s"), *ErrorText); }));
-	while (SpatialGDKEditor.IsSchemaGeneratorRunning())
-		FPlatformProcess::Sleep(0.1f);
-}
-
-void UGenerateSchemaAndSnapshotsCommandlet::GenerateSnapshots(FSpatialGDKEditor& SpatialGDKEditor)
-{
-	FString MapDir = TEXT("/Game");
-	UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Searching %s for maps"), *MapDir);
-	TArray<FString> MapFilePaths = GetAllMapPaths(MapDir);
-	for (FString MapFilePath : MapFilePaths)
+	//Massage input to allow some flexibility in command line path argument:
+	//	/Game/Path/MapName = Single map
+	//	/Game/Path/DirName/ = All maps in a dir, recursively
+	//NOTE: "/Game" is optional. If "/Game" is not included, the starting "/" is optional.
+	//Spaces in paths are DISALLOWED (it will currently break recursion, but there's no guarantee that
+	//will be consistent behaviour going forward).
+	//A single map is differentiated from a directory by the inclusion of "/" at the end of the path.
+	FString CorrectedPath;
+	if (!InPath.StartsWith(GameDirName))
```

Comment:
FString::AppendPath() is good for this situation - it ensures the correct folder symbols get added.